### PR TITLE
Added New Component: Speed dial

### PR DESCRIPTION
**Pull request by arpitBhalla**
**Summary**
Fixing #2880
https://material.io/components/buttons-floating-action-button#types-of-transitions
https://material-ui.com/components/speed-dial/#speed-dial

- [x] Component
- [x] Documentation
- [x] Tests

https://user-images.githubusercontent.com/55053424/112002725-35b5a380-8b46-11eb-95b0-673c0d5a535e.mp4

### CODE REVIEW

**Comment by arpitBhalla** [comment]
@flyingcircle Kindly Review

**Comment by flyingcircle** [comment]
@arpitBhalla this looks great! The only issue I see is that when I try it out on iOS, pressing anywhere on the screen opens the menu. That seems to be incorrect. Not sure if that's because of the demo app code or the component itself.

**Comment by arpitBhalla** [comment]
@flyingcircle Please Review

**Comment by arpitBhalla** [comment]
@flyingcircle Kindly Review

**Review by arpitBhalla — COMMENTED** [review]
- [ ] Done

File: src/buttons/SpeedDial.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      </View>
+    );
+  },
+  'SpeedDialAction'
+);
+
+export type SpeedDialProps = {
+  open?: boolean;
+  openIcon?: IconNode;
+  onChange?: () => void;
```
Comment (flyingcircle) [inline src/buttons/SpeedDial.tsx:33]: This seems unnecessary. Why not just `onPress`?

**Review by flyingcircle — CHANGES_REQUESTED** [review]

**Comment by arpitBhalla** [comment]
@flyingcircle Please Review

File: src/buttons/SpeedDial.tsx (hunk 1 above)
Comment (arpitBhalla) [reply src/buttons/SpeedDial.tsx]: onChange toggle the SpeedDial,

Comment (arpitBhalla) [reply src/buttons/SpeedDial.tsx:33]: Or `onClose`, `onOpen` can be used

Comment (flyingcircle) [reply src/buttons/SpeedDial.tsx:33]: Yeah that's confusing with an `open` prop. Is the openness controlled by the component or by `onChange`? If there's an `onOpen` and `onClose` then having an `open` prop doesn't make sense. OR If it's just `onPress` and having the developer control an `isOpen` `prop` would also make sense. But this seems to be trying to do both at once.

Comment (arpitBhalla) [reply src/buttons/SpeedDial.tsx:33]: SIr SpeedDial should be closed on backdrop press,

Comment (arpitBhalla) [reply src/buttons/SpeedDial.tsx:33]: https://material-ui.com/api/speed-dial/#props can I do according to Material UI component?

Comment (flyingcircle) [reply src/buttons/SpeedDial.tsx:33]: I suppose that should be fine.

**Comment by arpitBhalla** [comment]
@flyingcircle  Kindly Review

**Comment by arpitBhalla** [comment]
@flyingcircle Kindly Review

**Comment by arpitBhalla** [comment]
@flyingcircle Kindly Review

**Comment by flyingcircle** [comment]
I test in iOS and I still had a weird experience. When I had the menu open and I clicked to the left of a button in the Speed Dial, the menu didn't close, but if I clicked anywhere else on the screen, then it did close. It looks like there's a couple of transparent boxes that are blocking the backdrop Press from triggering.

**Comment by arpitBhalla** [comment]
> I test in iOS and I still had a weird experience. When I had the menu open and I clicked to the left of a button in the Speed Dial, the menu didn't close, but if I clicked anywhere else on the screen, then it did close. It looks like there's a couple of transparent boxes that are blocking the backdrop Press from triggering.

Do you mean SpeedDial Action Label?

**Comment by arpitBhalla** [comment]
> I test in iOS and I still had a weird experience. When I had the menu open and I clicked to the left of a button in the Speed Dial, the menu didn't close, but if I clicked anywhere else on the screen, then it did close. It looks like there's a couple of transparent boxes that are blocking the backdrop Press from triggering.

`elevation` might not work in iOS

**Comment by arpitBhalla** [comment]
@flyingcircle  I have tested in both android and iOS, I hope this time it will run smooth 😯

**Pull request merged**